editors/sc-ide/core/sc_lexer.cpp

```diff
@@ -153,12 +154,13 @@ Token::Type ScLexer::nextTokenInCode(int& lengthResult) {
 
         for (; it != end; ++it) {
             LexicalRule const& rule = *it;
-            int matchIndex = rule.expr.indexIn(mText, mOffset, QRegExp::CaretAtOffset);
-            // a guard to ensure all regexps match only at beginning of string:
-            Q_ASSERT(matchIndex <= mOffset);
-            if (matchIndex != -1) {
+            QRegularExpressionMatch match = rule.expr.match(mText, mOffset, QRegularExpression::NormalMatch,
+                                                            QRegularExpression::AnchoredMatchOption);
+            // a guard to ensure all regexps match only at the beginning of the string:
+            Q_ASSERT(match.hasMatch() && match.capturedStart() == mOffset);
```

Comment:
This assert fails if it doesn't match the first lexical rule. Should it go inside the following if statement?

Reply:
```
if (match.hasMatch()) {
    // a guard to ensure all regexps match only at the beginning of the string:
    Q_ASSERT(match.capturedStart() == mOffset);
    type = rule.type;
    length = match.capturedLength();
    break;
}
```

Good catch! Is this what you mean?

Reply:
sure, that is what I thought you meant in the first place.

Reply:
@capital-G this should fix that.

Reply:
Ah! I forgot to test it in debug build.